src/leader.py

```diff
@@ -152,6 +155,38 @@ def lead(self):
             self.turn(code, position)
 
 
+    def drive_in_circle(self):
+        """Drives in a circle."""
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.turn_left()
+
+
+    def drive_wide_winding(self):
+        """Drives in a wide, snake-like pattern."""
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.turn_right()
+        sleep(DEMO_TURN_TIME)
+
+        self.turn_left()
+        sleep(DEMO_TURN_TIME)
+
+
+    def drive_winding(self):
+        """Drives in a snake-like pattern."""
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.fw.turn(110)
+        sleep(DEMO_TURN_TIME)
+
+        self.fw.turn(70)
+        sleep(DEMO_TURN_TIME)
+
+
```

Comment:
We need a clean up function `stop_demo()` for the demos. This function can called when we're going out of demo mode to reset the car to its original state.